### PR TITLE
derive ‘elem’ from ‘equals’ and ‘reduce’

### PR DESCRIPTION
**Pull request by davidchambers**
This pull request is part of the effort to define "derived" `S` functions as `Z` functions.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1853,6 +1853,46 @@
...
+  //. false
+  //.
+  //. > elem(0, Nothing)
+  //. false
+  //. ```
+  function elem(x, foldable) {
+    //  Fast path for arrays.
+    if (Array.isArray(foldable)) {
+      return foldable.some(function(y) { return equals(x, y); });
+    }
```
Comment (davidchambers) [inline index.js]: @Avaq, @safareli, and others: are you in favour of this fast path? Having a fast path for `size` is a clear win, but for `elem` the fast path is no better than the general path in the worst case. Is this special case worthwhile?

Comment (safareli) [reply index.js]: Not sure, but if we had short circuited fold then we would have no need for this.

Comment (Avaq) [reply index.js]: I don't think it's worth it. Let the consumer optimise for specific data types. If I would be using `elem` on an array in a hot path I would probably optimise it out regardless of whether it has this fast path implemented.

Comment (kedashoe) [reply index.js]: > If I would be using elem on a significantly large array in a hot path I would probably optimise it out regardless of whether it has this fast path implemented.

If one is doing UI stuff, then sure, if you notice things slowing down you might do some optimizing of this sort. But I know for server stuff I write I probably wouldn't take the time even if maybe I should. It's only becoming more and more popular to spin up temporary instances to do one job and then die, so every millisecond costs money. I'd vote to leave it in :)

Comment (davidchambers) [reply index.js]: I find your comment persuasive, Aldwin. I removed the special case.

**Pull request merged**